conmon/conmon.c

```diff
@@ -116,6 +116,65 @@ int set_k8s_timestamp(char *buf, ssize_t buflen, const char *stream_type)
 	return 0;
 }
 
+
+/*
+ * splits buf into lines and inserts timestamps at the beginning of
+ * the line written to logfd.
+ */
+int write_with_timestamps(int logfd, const char *buf, ssize_t buflen)
+{
+	#define TSBUFLEN 34
+	char tsbuf[TSBUFLEN];
+	static bool last_buf_ended_with_newline = TRUE;
+
+	g_auto(GStrv) lines = g_strsplit(buf, "\n", -1);
+	ssize_t num_lines = g_strv_length(lines);
+	for (ssize_t i = 0; i < num_lines; i++)
+	{
+		const char *line = lines[i];
+
+		ninfo("Processing line: %ld, %s", i, line);
+
+		/* Skip last line if it is empty */
+		if (i == (num_lines - 1) && buf[buflen-1] == '\n' && !strcmp("", line)) {
+			ninfo("Skipping last line");
+			break;
+		}
+
+		/*
+		 * Only add timestamps for first line if last buffer's last
+		 * line ended with newline. Add it for all other lines.
+		 */
+		if (i != 0 || (i == 0 && last_buf_ended_with_newline)) {
+			ninfo("Adding timestamp");
+			int rc = set_k8s_timestamp(tsbuf, TSBUFLEN, "stdout");
```

Comment:
@runcom `ninfo("fuck %ld", TSBUFLEN+len);` sort of captures my opinion about having to do this in C :wink:. I think we could make this code "nicer" if we wrote partial lines to the log (so we just append the `timestamp` after the `\n` and don't do any of this code that stores the "remaining" part of the line). The downside is that a user might be able to see the partially-written lines. I don't think it should happen that often in practice, but you never know.

If you want, I'll open up a PR with my implementation of this so you can compare it with this.